ca/certificates.go

```diff
@@ -415,6 +390,100 @@ func NewRootCA(certBytes, keyBytes []byte, certExpiry time.Duration) (RootCA, er
 	return RootCA{Signer: signer, Key: keyBytes, Digest: digest, Cert: certBytes, Pool: pool}, nil
 }
 
+// ValidateCertChain checks that that a bundle certificates for a chain, each one certifying the one above,
+// as per RFC5246 section 7.4.2. They must also form at least one complete chain up to the root pool provided.
+// If everything validates, the entire parsed chain is returned.  Note that the chain can contain some unused
+// certificates (ones that aren't used to form a chain up to the given root pool).  This validation also enforces
+// that none of the certificates in the chain can be expired (unless allowExpired is set to true) or not yet
+// valid. This always returns at least 1 certificate.
+func ValidateCertChain(rootPool *x509.CertPool, certs []byte, allowExpired bool) ([]*x509.Certificate, error) {
```

Comment:
We don't have the root key material until the raft data is loaded, and before that gets loaded we get the TLS credentials first (either from disk, or from a remote cluster).  If the TLS cert is expired, starting the node fails and we can't load the key material in order to generate new ones.  Hence this change lets the node start up with expired certs, which it will then attempt to renew right away using the local signer once the raft data has been loaded and the CA server starts.

https://github.com/docker/swarmkit/pull/1919 was the original PR that added the the check to `verifyCertificate`.